subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaseCrossBuildResultsStore.java

```diff
@@ -260,6 +263,9 @@ public Void execute(Connection connection) throws SQLException {
             if (!DataBaseSchemaUtil.columnExists(connection, "TESTEXECUTION", "HOST")) {
                 statement.execute("alter table testExecution add column if not exists host varchar");
             }
+            if (!DataBaseSchemaUtil.columnExists(connection, "TESTEXECUTION", "TEAMCITYBUILDID")) {
+                statement.execute("alter table testExecution add column if not exists teamCityBuildId int not null default " + DEFAULT_TEAMCITY_BUILD_ID);
```

Comment:
Moreover, the type should be `varchar` I think, there are no guarantees that teamcity keeps that being integers, right?

Maybe the column also should be nullable so we can set it to null when we don't know the build id?

Reply:
It must be nullable, otherwise previous null buildId rows would complain.